File: src/backend/studio-reservation/src/main/java/org/taonaw/studio_reservation/domain/model/equipmentCategory/EquipmentCategory.java
package org.taonaw.studio_reservation.domain.model.equipmentCategory;

import lombok.NonNull;

import java.util.Objects;

public class EquipmentCategory {
    private EquipmentCategoryId id;
    private EquipmentCategoryName name;

    private EquipmentCategory(@NonNull EquipmentCategoryId id) {
        this.id = id;
    }

    public static EquipmentCategory create(@NonNull EquipmentCategoryName name) {
        var instance = new EquipmentCategory(EquipmentCategoryId.newId());
        instance.name = name;
        return instance;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        EquipmentCategory that = (EquipmentCategory) o;
        return id.equals(that.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }
}

File: src/backend/facility/src/main/kotlin/org/taonaw/facility/domain/model/equipmentCategory/EquipmentCategory.kt
package org.taonaw.facility.domain.model.equipmentCategory

class EquipmentCategory private constructor(
        val equipmentCategoryId: EquipmentCategoryId,
        val equipmentCategoryName: EquipmentCategoryName) {

    companion object {
        fun create(equipmentCategoryName: EquipmentCategoryName): EquipmentCategory {
            return EquipmentCategory(EquipmentCategoryId.newId(), equipmentCategoryName)
        }
    }

    fun change(equipmentCategoryName: EquipmentCategoryName): EquipmentCategory {
        return EquipmentCategory(equipmentCategoryId, equipmentCategoryName)
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as EquipmentCategory

        if (equipmentCategoryId != other.equipmentCategoryId) return false

        return true
    }

    override fun hashCode(): Int {
        return equipmentCategoryId.hashCode()
    }
}